### ISSUE TITLE
Model casts and toArray incompatibility

### ISSUE THREAD

**Issue by SeinopSys** [BenSampo/laravel-enum#162]
Let me start off by saying thank you for this very useful package! After using it for a bit I ran into an issue with using casts on models, namely, when I called `$user->toArray()` for example, with the user model having a cast to an enum defined in it, the return value of the casted model attribute ended up being an array that contained the name, value and description keys, instead of just the string value of the enum.

For the time being I'm using the following trait on all models that use such casts to overcome this, I haven't been able to find a better solution yet.

```php
<?php

namespace App\Traits;

use BenSampo\Enum\Enum;

trait HasEnumCasts
{
    public function toArray(): array
    {
        $result = parent::toArray();
        foreach ($result as &$value) {
            if ($value instanceof Enum) {
                $value = $value->value;
            }
        }
        return $result;
    }
}
```

**Comment by atymic** [comment]
This seems to be a change in Laravel 7, because it returns the cast value instead of the raw one like the package used to (it only overrode the getters).

I handle it in the API resource level, but i'm not sure we should override the `toArray` method (since it blocks any other packages that do the same).

Maybe we could add a readme note about updating the base model of your app to override the function?

**Comment by BenSampo** [comment]
@SeinopSys This should be fixed in v2.2.0. Can you upgrade and try again? Let me know if you're still having issues.

**Comment by SeinopSys** [comment]
It seems the fix worked, I'm no longer seeing the issue after the upgrade. Thank you!
